Apps/phcanary/canary_connector.py

```diff
@@ -0,0 +1,464 @@
+
+        # For now return Error with a message, in case of success we don't set the message, but use the summary
+        # return action_result.set_status(phantom.APP_ERROR, "Action not yet implemented")
+
+    def _handle_list_incidents(self, param):
+
+        # Add an action result object to self (BaseConnector) to represent the action for this param
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        # NOTE: test connectivity does _NOT_ take any parameters
+        # i.e. the param dictionary passed to this handler will be empty.
+        # Also typically it does not add any data into an action_result either.
+        # The status and progress messages are more important.
+        config = self.get_config()
+        apiKey = config.get('api_key')
+        params = {'auth_token': apiKey}
+
+        incidentState = param['incident_state']
+        if incidentState == "acknowledged":
+            endpoint = "/incidents/acknowledged"
+        if incidentState == "unacknowledged":
+            endpoint = "/incidents/unacknowledged"
+
+        self.save_progress("Connecting to endpoint")
+        # make rest call
+
+        ret_val, response = self._make_rest_call(endpoint, action_result, params=params, headers=None, method="get")
+
+        if (phantom.is_fail(ret_val)):
+            # the call to the 3rd party device or service failed, action result should contain all the error details
+            # for now the return is commented out, but after implementation, return from here
+            self.save_progress("Test Connectivity Failed.")
+            # return action_result.get_status()
+
+        if response:
+            action_result.add_data(response)
+            summary = action_result.update_summary({})
+            summary['count'] = len(response['incidents'])
+
+            if response.get('result') == "success":
+                return action_result.set_status(phantom.APP_SUCCESS)
+            else:
+                return action_result.set_status(phantom.APP_ERROR, "Error while communicating with Canary API")
+        else:
+            return action_result.set_status(phantom.APP_ERROR, "Error while communicating with Canary API")
+
+        # For now return Error with a message, in case of success we don't set the message, but use the summary
+        # return action_result.set_status(phantom.APP_ERROR, "Action not yet implemented")
+
+    def _handle_on_poll(self, param):
+
+        # Implement the handler here
+        # use self.save_progress(...) to send progress messages back to the platform
+        self.save_progress("In action handler for: {0}".format(self.get_action_identifier()))
+
+        # Add an action result object to self (BaseConnector) to represent the action for this param
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        config = self.get_config()
+        apiKey = config.get('api_key')
+        params = {'auth_token': apiKey}
```

Comment:
Params could be set in _make_rest_call() so its not repeated in every action handler

Reply:
@alexa-phantom We can do that. We do not have asset to verify after changing the code. Do you want me to change it?

Reply:
Yes please

Reply:
Fixed.